docs/DML-Operations-on-Carbon.md

```diff
@@ -1,197 +1,308 @@
+| number_of_segments | limit the output to this number. | YES |
+
+**Example:**
+
+  ```ruby
+  SHOW SEGMENTS FOR TABLE CarbonDatabase.CarbonTable LIMIT 2;
+  ```
+
+***
+
+# DELETE SEGMENT BY ID
+
+This command is to delete segment by using the segment ID.
+
+  ```ruby
+  DELETE SEGMENT segment_id1,segment_id2 FROM TABLE [db_name.]table_name;
+  ```
+
+### Parameter Description
+
+| Parameter | Description | Optional |
+| ------------- | -----| --------- |
+| segment_id | Segment Id of the load. | NO |
+| db_name | Database name, if it is not specified then it uses current database. | YES |
+| table_name | The name of the table in provided database.| NO |
+
+**Example:**
+
+  ```ruby
+  DELETE SEGMENT 0 FROM TABLE CarbonDatabase.CarbonTable;
+  DELETE SEGMENT 0.1,5,8 FROM TABLE CarbonDatabase.CarbonTable;
+  ```
+  Note: Here 0.1 is compacted segment sequence id.  
+
+***
+
+# DELETE SEGMENT BY DATE
+This command will allow to deletes the Carbon segment(s) from the store based on the date provided by the user in the DML command. The segment created before the particular date will be removed from the specific stores.
+
+  ```ruby
+  DELETE SEGMENTS FROM TABLE [db_name.]table_name WHERE STARTTIME BEFORE [DATE_VALUE];
+  ```
+
+### Parameter Description
+
+| Parameter | Description | Optional |
+| ------------- | -----| ------ |
+| DATE_VALUE | Valid segement load start time value. All the segments before this specified date will be deleted. | NO |
+| db_name | Database name, if it is not specified then it uses current database. | YES |
+| table_name | The name of the table in provided database.| NO |
+
+**Example:**
+
+  ```ruby
+  DELETE SEGMENTS FROM TABLE CarbonDatabase.CarbonTable WHERE STARTTIME BEFORE '2017-06-01 12:05:06';  
+  ```
+
+***
+
+# UPDATE CARBON TABLE
+This command updates the carbon table based on the column expression and optional filter conditions.
+ 
+The client node where the UPDATE command is executed should be part of the cluster.
+### Syntax
+Syntax1:
+  ```ruby
+  UPDATE <CARBON TABLE>
+  SET (column_name1, column_name2, ... column_name n) =  (column1_expression , column2_expression , column3_expression . .. column n_expression )
+  [ WHERE { <filter_condition> } ]; 
+  ```
+Syntax2:
+  ```ruby
+  UPDATE <CARBON TABLE>
+  SET (column_name1, column_name2,) =  (select sourceColumn1, sourceColumn2 from sourceTable [ WHERE { <filter_condition> } ] )
+  [ WHERE { <filter_condition> } ];
+  ```
+
+### Parameter Description
+
+| Parameter | Description |
+| ------------- | -----|
+| CARBON TABLE | The name of the Carbon table in which you want to perform the update operation. |
+| column_name | The destination columns to be updated. |
+| sourceColumn | The source table column values to be updated in destination table. |
+| sourceTable | The table from which the records are updated into destination Carbon table. |
+ 
+### Usage Guidelines
+Following are the conditions to use UPDATE:

```

Comment:
Here, please organize the description for syntax 1 and syntax 2 separately. For each section, basically we want to describe in what case the update will fail, and provide a brief reason that why it should failed